### PR TITLE
Add Reboot and Shutdown buttons to web interface #15

### PR DESCRIPTION
**Pull request by BoOsteous**
Hi,

This is my proposal for adding shutdown and reboot buttons to the web interface,
in the system information page.

I made the choice of adding a new shutdown method directly in the API of nabd,
to re-use the shutdown function already existing and doing everything in one single place...

Please tell me what do you think of it and if something need to be modified :)

Related to #15

### CODE REVIEW

**Comment by BoOsteous** [comment]
Will try to improve tests tomorrow to improve tests coverage

**Comment by BoOsteous** [comment]
image

**Comment by pguyot** [comment]
Thank you for this nice addition! I have yet to test and review your code. Judging from the screenshot, it seems you did use Javascript's built-in alert (?). Could you please use Bootstrap's dialog instead for consistency with ther other modal alerts (mostly for software update) ?

https://getbootstrap.com/docs/4.4/components/modal/

**Comment by BoOsteous** [comment]
> Thank you for this nice addition! I have yet to test and review your code. Judging from the screenshot, it seems you did use Javascript's built-in alert (?). Could you please use Bootstrap's dialog instead for consistency with ther other modal alerts (mostly for software update) ?
> 
> https://getbootstrap.com/docs/4.4/components/modal/

Yes i did :) i will change to bootstrap this evening no issue. thanks

**Comment by f-laurens** [comment]
Cosmetic interface change proposal:
Change  **Are you sure ? / Etes-vous sûr(e) ?**  prompt text in confirmation dialog to **Do you want to proceed? / Voulez-vous continuer?**

File: nabweb/locale/fr_FR/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -460,54 +460,90 @@ msgstr "Activé"
...
+#: templates/nabweb/system-info/index.html:99
 msgid "LEDs"
 msgstr "LEDs"
 
-#: templates/nabweb/system-info/index.html:81
+#: templates/nabweb/system-info/index.html:100
 msgid "Ears"
 msgstr "Oreilles"
 
+#: templates/nabweb/system-info/index.html:133
```
Comment (f-laurens) [inline nabweb/locale/fr_FR/LC_MESSAGES/django.po:535]: Suggestion for wording changes: See f-laurens@ee665ab.

File: nabweb/locale/fr_FR/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -460,54 +460,90 @@ msgstr "Activé"
...
+msgid "Reboot has been initiated !"
+msgstr "Le redémarrage est en cours !"
+
+#: templates/nabweb/system-info/index.html:37
+msgid "Shutdown has been initiated !"
+msgstr "L'arrêt est en cours !"
+
+#: templates/nabweb/system-info/index.html:37
+msgid "Please wait at least 10 seconds before unplugging the power cable"
+msgstr "Veuillez attendre au moins 10 secondes avant de débrancher le cable d'alimentation"
```
Comment (f-laurens) [inline nabweb/locale/fr_FR/LC_MESSAGES/django.po:485]: Suggestion for wording changes: see f-laurens/pynab@ee665ab.

File: nabweb/locale/fr_FR/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -460,54 +460,90 @@ msgstr "Activé"
...
+msgid "You are about to REBOOT the Raspberry Pi Linux system"
+msgstr "Vous êtes sur le point de REDEMARRER le système Linux du Raspberry Pi"
+
+#: templates/nabweb/system-info/index.html:135
+msgid "You are about to SHUTDOWN the Raspberry Pi Linux system"
+msgstr "Vous êtes sur le point d'ETEINDRE le système Linux du Raspberry Pi"
+
+#: templates/nabweb/system-info/index.html:137
+msgid "Are you sure ?"
+msgstr "Etes-vous sûr(e) ?"
```
Comment (f-laurens) [inline nabweb/locale/fr_FR/LC_MESSAGES/django.po:545]: Suggestion for wording changes: see f-laurens/pynab@ee665ab.

**Comment by mehdichaouch** [comment]
@BoOsteous are you able to rebase master, fix conflict, squash your last 3 commits and get @f-laurens changes ? Otherwise I can to do this 😉

**Comment by f-laurens** [comment]
To be closed as duplicate since PR #185 is now merged ?..

**Comment by mehdichaouch** [comment]
I tried both button, they work well.
Agreed with @f-laurens to close this PR.

**Comment by pguyot** [comment]
Already merged in #185

**Pull request closed without merging**